chandra_aca/manvr_mon_images.py

```diff
@@ -0,0 +1,195 @@
+        - sum_outlier: boolean flag for images with total sum outliers [slot]
+        - corr_sum_outlier: boolean flag for bgd-subtracted sum outliers [slot]
+        - bad_pixels: boolean flag for images with bad pixels [slot]
+        - t_ccd: CCD temperatures in Celsius
+        - idx_manvr: maneuver index for each sample
+
+    Notes
+    -----
+    - Raw images in DN are converted to e-/s using factor 5.0/1.696
+    - Temperature correction uses dark_temp_scale() from chandra_aca.dark_model
+    - Early CCD temperature samples are replaced with 5th sample to avoid artifacts
+    """
+    start = CxoTime(start)
+    stop = CxoTime(stop)
+    data_path = imgs_root_dir_path(data_dir)
+
+    imgs_list = []
+    masks_list = []
+    times_list = []
+    row0s_list = []
+    col0s_list = []
+    t_ccds_list = []
+    earth_limb_angles_list = []
+    idx_manvrs_list = []
+    idx_manvr = 0
+
+    for year, doy in get_years_doys(start, stop):
+        dir = Path(data_path, year, doy)
+        for path in sorted(dir.glob("*.npz")):
+            with np.load(path) as dat:
+                n_samp = len(dat["t_ccd"])
+                # Move the sample dimension (-1) to the front so they concat properly.
+                # Imgs will then be indexed as [sample, slot, row, col].
+                imgs_list.append(np.moveaxis(dat["slot_imgs"], -1, 0))
+                # Masks will be [sample, slot].
+                masks_list.append(np.moveaxis(dat["slot_masks"], -1, 0))
+                # Repeat slot_row0s n_samp times for each sample.
+                row0s_list.append(np.repeat(dat["slot_row0s"][None, :], n_samp, axis=0))
+                col0s_list.append(np.repeat(dat["slot_col0s"][None, :], n_samp, axis=0))
+                t_ccds = dat["t_ccd"]
+                t_ccds[:5] = t_ccds[5]
+                t_ccds_list.append(t_ccds)
+                earth_limb_angles_list.append(dat["earth_limb_angle"])
+                times_list.append(dat["time0"] + 4.1 * np.arange(n_samp))
+                idx_manvrs_list.append(idx_manvr + np.zeros(n_samp, dtype=np.int32))
+                idx_manvr += 1
+
+    dat = apt.Table()
+
+    dat["time"] = np.concatenate(times_list)
+    dat["img_raw"] = np.concatenate(imgs_list)
+    masks = np.concatenate(masks_list)
+    dat["mask"] = masks
+    dat["sum_outlier"] = (masks & ImgStatus.SUM_OUTLIER.value) != 0
+    dat["corr_sum_outlier"] = (masks & ImgStatus.CORR_SUM_OUTLIER.value) != 0
+    dat["bad_pixels"] = (masks & ImgStatus.HAS_BAD_PIX.value) != 0
+    dat["t_ccd"] = np.concatenate(t_ccds_list)
+    dat["earth_limb_angle"] = np.concatenate(earth_limb_angles_list)
+    dat["idx_manvr"] = np.concatenate(idx_manvrs_list)
+    dat["row0"] = np.concatenate(row0s_list)
+    dat["col0"] = np.concatenate(col0s_list)
+
+    ok = (dat["time"] >= start.secs) & (dat["time"] < stop.secs)
+    dat = dat[ok]
+
+    # Make idx_manvr start at 0
+    dat["idx_manvr"] -= dat["idx_manvr"][0]
```

Comment:
Seems like this will fail if there are no ok values above, but maybe we don't care at that point.